src/Kris/LaravelFormBuilder/Form.php

```diff
@@ -192,6 +214,38 @@ public function addAfter($name, $fieldName, $type = 'text', $options = [], $modi
 
         return $this;
     }
+    
+    /**
+     * Take another form and add it's fields directly to this form
+     * @param mixed   $class        Form to merge
+     * @param array   $options
+     * @param boolean $modify
+     * @return $this
+     */
+    public function compose($class, array $options = [], $modify = false)
+    {
+        $options['class'] = $class;
+
+        // If we pass a ready made form just extract the fields
+        if ($class instanceof Form) {
+            $fields = $class->getFields();
+        } elseif ($class instanceof Fields\ChildFormType) {
+            $fields = $class->getForm()->getFields();
+        } elseif (is_string($class)) {
+            // If its a string of a class make it the usual way
+            $childForm = $this->makeField('', 'form', $options);
```

Comment:
Instead of making ChildForm, you could create regular form instance here like this:
```php
$options['model'] = $this->model;
$options['name'] = $this->name;

$form = $this->formBuilder->create($class, $options);
$fields = $form->getFields();
```

I don't like the idea of creating fields with empty names, even if only parts are used. This solves the problem.